src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -8413,617 +9036,195 @@ void SketchObject::rebuildExternalGeometry(bool defining, bool addIntersection)
                     sync = true;
+                }
             }
         }
         if(frozen && !sync) {
             refSet.insert(std::move(key));
             continue;
         }
-        if(!Obj || !Obj->getNameInDocument())
+        if (!Obj || !Obj->getNameInDocument()) {
             continue;
+        }
 
         std::vector<std::unique_ptr<Part::Geometry> > geos;
 
+        auto importVertex = [&](const TopoDS_Shape& refSubShape) {
+            gp_Pnt P = BRep_Tool::Pnt(TopoDS::Vertex(refSubShape));
+            GeomAPI_ProjectPointOnSurf proj(P, gPlane);
+            P = proj.NearestPoint();
+            Base::Vector3d p(P.X(), P.Y(), P.Z());
+            invPlm.multVec(p, p);
+
+            Part::GeomPoint* point = new Part::GeomPoint(p);
+            GeometryFacade::setConstruction(point, true);
+            geos.emplace_back(point);
+        };
+
         try {
-        TopoDS_Shape refSubShape;
+            TopoDS_Shape refSubShape;
 
-        if (Obj->isDerivedFrom<Part::Datum>()) {
-            const Part::Datum* datum = static_cast<const Part::Datum*>(Obj);
-            refSubShape = datum->getShape();
-        }
-        else if (Obj->isDerivedFrom<Part::Feature>()) {
-            try {
-                const Part::Feature* refObj = static_cast<const Part::Feature*>(Obj);
-                const Part::TopoShape& refShape = refObj->Shape.getShape();
-                refSubShape = refShape.getSubShape(SubElement.c_str());
+            if (Obj->isDerivedFrom<Part::Datum>()) {
+                auto* datum = static_cast<const Part::Datum*>(Obj);
+                refSubShape = datum->getShape();
             }
-            catch (Standard_Failure& e) {
-                throw Base::CADKernelError(e.GetMessageString());
+            else if (Obj->isDerivedFrom<Part::Feature>()) {
+                try {
+                    auto* refObj = static_cast<const Part::Feature*>(Obj);
+                    const Part::TopoShape& refShape = refObj->Shape.getShape();
+                    refSubShape = refShape.getSubShape(SubElement.c_str());
+                }
+                catch (Standard_Failure& e) {
+                    throw Base::CADKernelError(e.GetMessageString());
+                }
+            }
+            else if (Obj->isDerivedFrom<App::Plane>()) {
+                auto* pl = static_cast<const App::Plane*>(Obj);
+                Base::Placement plm = pl->Placement.getValue();
+                Base::Vector3d base = plm.getPosition();
+                Base::Rotation rot = plm.getRotation();
+                Base::Vector3d normal(0, 0, 1);
+                rot.multVec(normal, normal);
+                gp_Pln plane(gp_Pnt(base.x, base.y, base.z), gp_Dir(normal.x, normal.y, normal.z));
+                BRepBuilderAPI_MakeFace fBuilder(plane);
+                if (!fBuilder.IsDone())
+                    throw Base::RuntimeError(
+                        "Sketcher: addExternal(): Failed to build face from App::Plane");
+

```

Comment:
Please encase within braces.

Reply:
If you move the code, why not format it? Especially as this can be done automatically.